spec/exmples/contacts_accessor_spec.rb

```diff
@@ -0,0 +1,65 @@
+require File.expand_path(File.dirname(__FILE__) + '../../spec_helper')
+
+describe GoogleContactsApi::ContactsAccessor do
+    before(:all) do
+      @contact_json_hash = contact_json_hash
+      @contact = GoogleContactsApi::Contact.new(@contact_json_hash)
+    end
+
+    let(:api) { double("api") }
+    let(:test_class) {
+      Class.new do
+        include GoogleContactsApi::ContactsAccessor
+        def initialize(api)
+          @api = api
+        end
+      end
+    }
+    describe ".get_contacts" do
+      it "should get the contacts using the internal @api object" do
+        expect(api).to receive(:get).with("contacts/default/full", kind_of(Hash)).and_return(Hashie::Mash.new({
+          "body" => "some response",
+          "code" => 200
+        }))
+        allow(GoogleContactsApi::ContactSet).to receive(:new).and_return("contact set")
+        expect(test_class.new(api).get_contacts).to eq("contact set")
+      end
+    end
+
+    # if they actually return different succeess codes I can validate on them too
+    describe ".put_or_post_contact" do
+      it "should put the contact when it exists" do
+        tc = test_class.new(@api)
```

Comment:
Well both of them aren't really "test classes", right? It's probably a `StubbedContactsAccessor` or something.

`stubbed_contacts_accessor = stubbed_contacts_accessor_class.new(@api)`